_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+
+Each branch is either a `String` or an `Object` with the following properties:
+
+| Option       | Description                                                                                                                                                             |
+|--------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `branch`     | **Require** Git branch name                                                                                                                                             |
+| `range`      | Accepted version range for to release from this branch. If defined the branch will be considered a LTS branch.                                                          |
+| `channel`    | Channel on which to release. Ignored for release targets that don't support channels.                                                                                   |
+| `prerelease` | Version of the pre-release done from this branch. Must be formatted `<version>-<tag>`. The version is required, the `tag` is optional. Examples: `4.0.0-beta`, `4.0.0`. |
+
+**The order in which the branches are defines is meaningful, and it is used to determine the branch type and which release can be done from each branch.**
+
+##### Branches naming convention
+
+In a branch is defined as a `String` `semantic-release` will automatically determine the type of branch as follow:
+- If the branch name is a valid semver range (Only the format `1.x.x` or `1.x` are valid Git reference names) the branch will be considered a LTS branch, with its range based on the branch name, and release on the channel with the same name
+- The first branch in the `branches` `Array` after the LTS branches is considered the default branch (usually `master`).
+- Each branches in the `branches` `Array` after the default branch are considered future branches.
+
+For example:
+```json
+{
+  "release": {
+    "branches": [
+      "1.x.x",
+      "master",
+      "next",
+      {"branch": "alpha", "prerelease": true}
+    ]
+  }
+}
+```
+is equivalent to:
+```json
+{
+  "release": {
+    "branches": [
+      {"branch": "1.x.x", "range": "1.x.x", "channel": "latest"},
+      {"branch": "master", "channel": "latest"},
+      {"branch": "next", "channel": "next"},
+      {"branch": "alpha", "channel": "alpha", "preid": "alpha", "prerelease": true}
+    ]
+  }
+}
+```
+
+##### Configuration validation
+
+The `branches` configuration must follow these rules:
+- All branch name must be valid [Git reference](https://git-scm.com/docs/git-check-ref-format)
+- A branch cannot be named after a valid semver version (for example `1.0.0` or `v1.0.0` are invalid branch names) or a name that match the `tagFormat`.
+- A default branch is required.
+- LTS branches range must not overlap among themselves and the default one. For example `["1.x.x", 1.5.x]` is invalid.
```

Comment:
this is no longer a problem if we only support the `1.x` format

Reply:
ah wait, people can configure custom ranges, so this is still a valid point. I wonder if should not allow for that to start out, it will make the implementation easier. Basically valid "range" values would only be `1.x`, `2.x`, etc

Reply:
That would not really make the implementation easier as we need to verify the LTS do not overlap with `master` and `next`. 
If the latest version on `master` is 2.1.1, we can't make releases 2.x.x releases from another branch. So we need to verify that if current release line on `master` is 2.x.x, there is no 2.x, 3.x, 4.x etc LTS branches.
The easiest way to do that is:
- check there is no overlap
- check the ranges corresponding to each branch are incremental (as second branch has a higher range than fist, the third has a higher range then the second etc)